src/Appwrite/Auth/OAuth2/Keycloak.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+namespace Appwrite\Auth\OAuth2;
+
+use Appwrite\Auth\OAuth2;
+
+// Reference Material
+// https://www.keycloak.org/docs/latest/securing_apps/index.html#other-openid-connect-libraries
+
+class Keycloak extends OAuth2
+{
+    /**
+     * @var array
+     */
+    protected array $scopes = [
+        'openid',
+        'profile',
+        'email',
+        'offline_access'
+    ];
+
+    /**
+     * @var array
+     */
+    protected array $user = [];
+
+    /**
+     * @var array
+     */
+    protected array $tokens = [];
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return 'keycloak';
+    }
+
+    /**
+     * @return string
+     */
+    public function getLoginURL(): string
+    {
+        return 'https://' . $this->getKeycloakDomain() . '/realms/' . $this->getKeycloakRealm() . '/protocol/openid-connect/auth?' . \http_build_query([
```

Comment:
Perhaps it would be useful to pass the scheme too for cases where you're testing kecloak and don't have a valid SSL certificate?

Reply:
What sounds better?

- One Parameter for host and protocol
- Two Seperate Parameter for host and protocol

Reply:
Let's go with 1 since that's how it's done for the Gitlab one:

https://github.com/appwrite/appwrite/blob/e53defb24afb100194b98168019dc84a468dcc1c/src/Appwrite/Auth/OAuth2/Gitlab.php#L42

Reply:
Ok I will change, maybe we should then also change Authentik

Reply:
done